pim-tool-back/src/main/java/vn/elca/training/web/ProjectsController.java

```diff
@@ -0,0 +1,37 @@
+package vn.elca.training.web;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import vn.elca.training.model.dto.ProjectDto;
+import vn.elca.training.service.ProjectService;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * @author gtn
+ * @author thomas.dang
+ */
+@RestController
+@RequestMapping("/projects")
+public class ProjectsController extends AbstractApplicationController {
+
+    @Autowired
+    private ProjectService projectService;
+
+    @GetMapping("/search")
+    public List<ProjectDto> search(@RequestParam String keyword) {
+        if (keyword.isBlank()) {
+            return new ArrayList<>();
+        }
+        return projectService.findAll()
+                .stream()
+                .filter(product -> product.getName().trim().toLowerCase()
```

Comment:
This is OK for the exercise because you did not learn JPA yet. But in pilot project, this filter should be performed at database level.